### ISSUE TITLE
Issue with some models when defaultLimit is 0 or -1 and null array

### ISSUE THREAD

**Issue by dmitrye** [balderdashy/sails-disk#32]
Set blueprints.js to include:
 
 defaultLimit: -1 

Or via API call example (limit=0): http://localhost:1337/transactions/find?Paid=false&limit=0

Run a query against certain controller default /find method

Resulting error:

error: Sending 500 ("Server Error") response: 
 TypeError: Cannot read property 'length' of null
    at Object.async.eachSeries ({{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/node_modules/async/lib/async.js:144:17)
    at {{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/cursor/populateBuffers.js:53:15
    at {{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/node_modules/async/lib/async.js:125:13
    at Array.forEach (native)
    at _each ({{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/node_modules/async/lib/async.js:46:24)
    at Object.async.each ({{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/node_modules/async/lib/async.js:124:9)
    at Array.async.auto.buildChildBuffers [as 1] ({{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/cursor/populateBuffers.js:44:13)
    at {{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/node_modules/async/lib/async.js:487:38
    at Array.forEach (native)
    at _each ({{..my app path ..}}/node_modules/sails-disk/node_modules/waterline-cursor/node_modules/async/lib/async.js:46:24) [TypeError: Cannot read property 'length' of null]

Two issues here:

1) On Line 144 of async.js the test is if(!arr.length) which blows up when the value of arr is null. A better condition that will not break is if(!!arr). 

2) When limit is 0 or defaultLimit -1 but collection has items it should not be returning a null arr object.

**Comment by blah238** [comment]
Thanks for posting this, I am also seeing this behavior with 0.10.5. I have set defaultLimit to -1 and am using sails-disk.

**Comment by sailsbot** [comment]
Thanks for posting, @dmitrye.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
